static/js/tools/timeline_util.ts

```diff
@@ -274,6 +274,72 @@ function setSearchParam(vars: VarUrl) {
   window.location.hash = newHash;
 }
 
+interface StatsVarNode {
+  [key: string]: string[][];
+}
+
+class UrlTimeline {
+  statsVarNodes: StatsVarNode;
+  placeDcids: string[];
+  pc: boolean;
+
+  constructor() {
+    this.statsVarNodes = {};
+    this.placeDcids = [];
+    this.pc = false;
+  }
+
+  public setPC(): void {
+    this.pc = true;
+  }
+
+  public unsetPC(): void {
+    this.pc = false;
+  }
+
+  public addPlace(placeDcid: string): void {
+    this.placeDcids.push(placeDcid);
+  }
+
+  public removePLace(placeDcid: string): void {
+    const index = this.placeDcids.indexOf(placeDcid);
+    if (index > -1) {
+      this.placeDcids.splice(index, 1);
+    }
+  }
+
+  public addStatsVar(statsVar: string): void {
+    if (!(statsVar in this.statsVarNodes)) {
+      this.statsVarNodes[statsVar] = [];
+    }
+  }
+
+  public addStatsVarWithPath(statsVar: string, nodePath: string[]): void {
+    if (!(statsVar in this.statsVarNodes)) {
+      this.statsVarNodes[statsVar] = [nodePath];
+    } else if (!this.statsVarNodes[statsVar].includes(nodePath)) {
```

Comment:
this might not work: try type "[[3,2],[4,5]].includes([3,2])" in chrome console.

try to add a test case for it

Reply:
fixed